src/neural/loader.cc

```diff
@@ -46,25 +50,89 @@ FloatVectors LoadFloatsFromFile(const std::string& filename) {
     } else {
       bytes_read += sz;
       buffer.resize(bytes_read);
-      // Add newline in the end for the case it was not there.
-      buffer.push_back('\n');
       break;
     }
   }
   gzclose(file);
 
+  return buffer;
+}
+
+void DenormLayer(const lc0::Weights_Layer& layer, FloatVectors& vecs) {
```

Comment:
Everything is in namespace `lczero`, maybe move proto also there rather than `lc0`?

Style guide doesn't allow non-const references! :)
Could you replace it with pointers?

Reply:
If I move proto there, it will clash with current lczero::Weights definition. So keeping as is for now.